### ISSUE TITLE
zookeeper leader log file grows and fills up disk space

### ISSUE THREAD

**Issue by rdhabalia** [apache/pulsar#4276]
### Issue

When Bk-client deletes the ledger, it also tries to delete ledger's parent nodes blindly and because of that ZK's entire log file fills up  with below log `Directory not empty for`. Broker rollovers cursor's metadata-ledger at every few configured hours. Broker with large number of global topics and cursors it generates really huge log (>10GB/day) file at zk-leader which consumes most of the disk space.

```
Tue May 14 00:00:00 2019: 2019-05-14 00:00:00,297 - INFO  [ProcessThread(sid:5 cport:-1)::PrepRequestProcessor@645] - Got user-level KeeperException when processing sessionid:0x123454567 type:delete cxid:0xc655f6 zxid:0x123454567 txntype:-1 reqpath:n/a Error Path:/ledgers/000/0000/0123/4567 Error:KeeperErrorCode = Directory not empty for /ledgers/000/0000/0123/4567
```

### Possible fix
1. Logging configuration change at ZK/Change log level but then it will prevent us to store important logs when it needs for troubleshooting.
2. Better solution :  BK-client should only delete parent node if it's empty. It also prevents unexpected data-loss due to unseen zk's bug while deleting non-empty znode.

**Comment by rdhabalia** [comment]
@sijie @merlimat does it make sense to check empty node before deleting ledger's parent node?

**Comment by merlimat** [comment]
Also, the "delete" operation will get appended to the ZK transaction log anyways, because it works as a write-ahead-log, so it's one more write op.

(Though I wouldn't characterize this as a "bug" :) )

**Comment by merlimat** [comment]
As a temp workaround, to increase log level to "warn" on `PrepRequestProcessor`. For proper fix, check the parent folder "stat" to get the list of current children and avoid issuing the delete requests.

**Comment by rdhabalia** [comment]
@merlimat can you please review: https://github.com/apache/bookkeeper/pull/2097

**Comment by merlimat** [comment]
LGTM

**Comment by sijie** [comment]
This is fixed by apache/bookkeeper#2097
